### DISCUSSION TITLE
How to load models and metadata at startup ?

### DISCUSSION THREAD

**Discussion by curiousmind88** [fastapi/fastapi#8894] [Questions]
I wanted to load a model and some metadata at startup in memory and use it for all subsequent requests. 
And I need to access this loaded data at startup across differnt modules.  

I was not able to find good resources on this.  

Could you please help what would be the best way to achieve this using FastAPI ?

**Comment by sathoune** [comment]
Do you mean machine learning model?

You probably can use cache:
```python
import functools

@functools.cache
def load_model():
    model = ```The loading logic```
    return model
```
Only the first call should take time and actually load the file. Subsequent calls will have very little time cost.

You can read more about the function here:
https://docs.python.org/3/library/functools.html#functools.cache

**Comment by curiousmind88** [comment]
Cool thanks, will check this out !

**Comment by ycd** [comment]
See @raphaelauv's comment on https://github.com/tiangolo/fastapi/issues/2425#issuecomment-734790381

**Comment by raphaelauv** [comment]
@curiousmind88 could you close the issue , thanks
